### ISSUE TITLE
Inconsistent `"use strict"` handling in the checker vs emitter

### ISSUE THREAD

**Issue by nicolo-ribaudo** [microsoft/TypeScript#59499]
### 🔎 Search Terms

use strict type emitter

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

https://www.typescriptlang.org/play/?alwaysStrict=false&target=1&module=0#code/C4TwDgpgBAhlC8UCMBuAUAIgK4Gdo+ACcBLAY2A3QDcZCoAbCYBKAJhSA

### 💻 Code

```ts
type a = 1;
"use strict";
var let = 2;
```

### 🙁 Actual behavior

TypeScript parses this code as loose mode, and thus does not report that `let` is not a valid identifier in strict mode. This is in line with what I would expect, since there is something before the `"use strict";` expression statement and thus it's not a directive.

However, TypeScript emits this code:
```js
"use strict";
var let = 2;
```

This code is in strict mode, because there is nothing before `"use strict"`.

### 🙂 Expected behavior

Either that code is strict (so TS should report an error about using `let` as an identifier) or it's not (so TS shouldn't emit it as strict mode code).

I think the most intuitive behavior is that that code is _not_ strict, and thus TS should emit it as
```js
;
"use strict";
var let = 2;
```
or as
```js
("use strict");
var let = 2;
```

### Additional information about the issue

This was originally mentioned on twitter by @acutmore as a "TypeScript weirdness", but I believe it is a bug as the checker and the emitter disagree.

Also, Babel has the same bug.

**Comment by RyanCavanaugh** [comment]
All module code is strict-mode anyway and this is a pretty difficult mistake to make in global code. Doesn't really seem worth further thought.
